Python/gc_free_threading.c

```diff
@@ -1952,25 +1952,25 @@ int
 PyGC_Enable(void)
 {
     GCState *gcstate = get_gc_state();
-    int old_state = gcstate->enabled;
-    gcstate->enabled = 1;
+    int old_state = _Py_atomic_load_int(&(gcstate->enabled));
+    while (!_Py_atomic_compare_exchange_int(&gcstate->enabled, &old_state, 1));
     return old_state;
 }
 
 int
 PyGC_Disable(void)
 {
     GCState *gcstate = get_gc_state();
-    int old_state = gcstate->enabled;
-    gcstate->enabled = 0;
+    int old_state = _Py_atomic_load_int(&(gcstate->enabled));
+    while (!_Py_atomic_compare_exchange_int(&gcstate->enabled, &old_state, 0));
```

Comment:
We don't need to update old_state explicitly since __atomic_compare_exchange_n automatically handle.
https://gcc.gnu.org/onlinedocs/gcc-8.4.0/gcc/_005f_005fatomic-Builtins.html